cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -114,6 +114,11 @@ func (g *gitHubRepository) GetFile(version, path string) ([]byte, error) {
 	return files, nil
 }
 
+// newGitHubRepository returns a gitHubRepository implementation.
+func NewGitHubRepository(providerConfig config.Provider, configVariablesClient config.VariablesClient, opts ...githubRepositoryOption) (Repository, error) {
```

Comment:
Why not making the existing method public?

Reply:
It returns a private struct pointer, not an interface.